src/smt/theory_str_noodler/formula_preprocess.cpp

```diff
@@ -1158,8 +1163,7 @@ namespace smt::noodler {
                         }
                     }
                 }
-            }
-            if(this->formula.single_occurr(pr.second.get_right_set())
+            } else if(this->formula.single_occurr(pr.second.get_right_set())
```

Comment:
Why is there the else branch? The first if may be true, but the inner conditions might be false; then it could go to the second if (which is now not possible).

Reply:
Ah, I didn't notice there is inner condition, I'll fix it.

Reply:
Fixed, I added `continue` instead.